practices/Git.md

```diff
@@ -0,0 +1,110 @@
+But Git on the other hand saves the data in the form of snapshots of the file-system. Whenever you commit some changes to the project it will take the snapshot and save it with a reference and for efficiency, it does not take the snapshot if the file is not changed it just links it to the previous ones.
+It was first created by [Linus Torvalds](https://en.wikipedia.org/wiki/Linus_Torvalds) in 2005 for the development of the [Linux kernel](https://en.wikipedia.org/wiki/Linux_kernel).
+
+
+## How Does Git Work?
+
+![Git_understanding](/files/Git_i_dont_know.png)</br>
+Nvie: [Git understanding](https://medium.com/hackernoon/https-medium-com-zspajich-understanding-git-data-model-95eb16cc99f5)
+
+Git is a lot more complicated than its looks at it's front-end command line. Let's have a look at it.
+
+#### Data Model
+
+When you make a repository, do some changes, and create a commit, Git generates a 40-Character checksum (SHA1) hash. These hashes are created for every object and the first two characters of the checksum are for the directory name and the rest of them are for the file name. 
+There are three types of objects created by Git. 
+- First, is the **blob** object which stores the snapshot of the files that we committed (there is a set of n objects for n committed files). 
+- Second, it creates the **tree** object which contains the list of all committed files having a pointer to blob objects. 
+- The third and final object is the **commit** object that has the pointer to the tree object.
+
+#### Branching
+
+A **branch** is an independent line of development for a particular project with its commits. How does Git does the branching? On the surface, it looks like a list of commits. However, Git only tracks the latest commit on the branch. Knowing the newest change, Git can easily reconstruct the whole commit list. A pointer named **head** which has a reference to the checkout branch (checkout branch tells Git to record all new commits on that branch) and is also used for tracking. 
+
+#### Index
+
+According to the Git's point of view, it has mainly three areas: **Working directory, staging area, and the repository**. When we work on a project on our local machine, Git stores the changes in your computer's filesystem is **working directory**. The changes that are made to the project will remain in the working directory unless we add them to the staging area using the `git add` command. The staging area describes the initial state of the new commit. You can use the staging area to fine-tune the commits by adding or removing until it suits you, after which you can do `git commit` to save it to the directory as an object. For this process, Git doesn't use any directory but on the other hand, it uses an **index** which can be used to tracks changes.
+
+## Why Git Is Preferred
```

Comment:
The third headline should be `Why You Might Want the %{name}`. See [Contribution guide](https://github.com/DXHeroes/knowledge-base-content/blob/master/CONTRIBUTING.md).

```suggestion
## Why You Might Want the Git
```